packages/vue/src/components/setup-totp.vue

```diff
@@ -1,44 +1,37 @@
 <script setup lang="ts">
-import { onMounted, reactive, computed, ComputedRef, useAttrs, ref } from 'vue';
+import { computed, ref, toRefs, useAttrs, onMounted, reactive } from 'vue';
 import QRCode from 'qrcode';
 
 import { Logger } from 'aws-amplify';
 import {
   authenticatorTextUtil,
-  getActorState,
   getFormDataFromEvent,
-  SignInState,
   translate,
   getTotpCodeURL,
 } from '@aws-amplify/ui';
 
-import { useAuth, useAuthenticator } from '../composables/useAuth';
+import { useAuthenticator } from '../composables/useAuth';
+import { UseAuthenticator } from '../types';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
 const logger = new Logger('SetupTOTP-logger');
 
-const props = useAuthenticator();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as UseAuthenticator;
+const { updateForm, submitForm, toSignIn } = props;
+const { error, isPending, QRFields , totpSecretCode, user } = toRefs(props);
```

Comment:
🤷 , this is why I shouldn't code directly on GitHub
```suggestion
const { error, isPending, QRFields, totpSecretCode, user } = toRefs(props);
```